test/functional/suite_test.go

```diff
@@ -47,6 +47,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 
```

Comment:
probably should add some tests for adding/removing the topologyref to see that the statefulset gets updated properly when move from default to toporef and back?

Reply:
Yes, the current tests only prove that we have no regression and 100% backward compatibility we the previous CR versions and the behavior is preserved (we don't get any unexpected result). I agree that adding / removing a `topologyRef` to actually test the new code introduced both here and in lib-common would be a nice improvement here. Let me experiment with that and I will update this patch.

Reply:
Done